src/pydata_sphinx_theme/__init__.py

```diff
@@ -819,6 +823,78 @@ def _overwrite_pygments_css(app, exception=None):
         f.write(get_pygments_stylesheet(light_theme, dark_theme))
 
 
+# ------------------------------------------------------------------------------
+# customize rendering of the links
+# ------------------------------------------------------------------------------
+
+
+class LinkTransform(SphinxPostTransform):
+    """
+    Shorten link when they are coming from github or gitlab and add an extra class to the tag
+    for further styling.
+    Before::
+        <a class="reference external" href="https://github.com/2i2c-org/infrastructure/issues/1329">
+            https://github.com/2i2c-org/infrastructure/issues/1329
+        </a>
+    After::
+        <a class="reference external github" href="https://github.com/2i2c-org/infrastructure/issues/1329">
+            2i2c-org/infrastructure#1329
+        </a>
+    """  # noqa
+
+    default_priority = 400
+    formats = ("html",)
+    supported_platform = {"github.com": "github", "gitlab.com": "gitlab"}
+    platform = None
+
+    def run(self, **kwargs):
+        matcher = NodeMatcher(nodes.reference)
+        for node in self.document.findall(matcher):
+            uri = node.attributes.get("refuri")
+            text = next(iter(node.children), None)
+            # only act if the uri and text are the same
+            # if not the user has already customized the display of the link
+            if uri is not None and text is not None and text == uri:
```

Comment:
Will this work if people use link syntax like: `<https://github.com>`?

Reply:
I tested it with `<https://github.com/2i2c-org/infrastructure/issues/1329>` and it worked. As I'm testing the already parsed node, I am agnostic on how the link is generated